planetscale/database_branch_resource.go

```diff
@@ -236,3 +239,64 @@ func (r *databaseBranchResource) Configure(_ context.Context, req resource.Confi
 
 	r.client = req.ProviderData.(*planetscale.Client)
 }
+
+func splitDatabaseBranchResourceID(id string) (teamID, _id string, branchName string, ok bool) {
+	attributes := strings.Split(id, "/")
+	requiredAttributesLength := 3
+	if len(attributes) != requiredAttributesLength {
+		return "", "", "", false
+	}
+	return attributes[0], attributes[1], attributes[2], true
+}
+
+func (r *databaseBranchResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	organizationName, databaseName, branchName, ok := splitDatabaseBranchResourceID(req.ID)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Error importing database",
+			fmt.Sprintf("Invalid id '%s' specified. should be in format \"organization_name/database_name/branch_name\"", req.ID),
+		)
+		return
+	}
+
+	out, err := r.client.DatabaseBranches.Get(ctx, &planetscale.GetDatabaseBranchRequest{
+		Organization: organizationName,
+		Database:     databaseName,
+		Branch:       branchName,
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading project",
```

Comment:
Not sure if the notion of "project" is appropriate use here, mind changing to something explicitly specific, eg "database branch password"?